### PR TITLE
Add go dir and move sample

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
- add go dir
- move Memcached go sample

### CODE REVIEW

File: go/memcached-operator/test/e2e/memcached_test.go [hunk 1]
```diff
@@ -17,12 +17,11 @@ package e2e
 import (
 	goctx "context"
 	"fmt"
+	apis2 "github.com/operator-framework/operator-sdk-samples/go/memcached-operator/pkg/apis"
```
Comment (estroz) [inline go/memcached-operator/test/e2e/memcached_test.go]: These imports should not be changed except for adding `go/` to the path.

File: go/memcached-operator/pkg/controller/memcached/memcached_controller_test.go [hunk 2]
```diff
@@ -2,13 +2,12 @@ package memcached
 
 import (
 	"context"
+	"github.com/operator-framework/operator-sdk-samples/go/memcached-operator/pkg/apis/cache/v1alpha1"
```
Comment (estroz) [inline go/memcached-operator/pkg/controller/memcached/memcached_controller_test.go]: These imports should not be changed except for adding `go/` to the path.

File: go/memcached-operator/pkg/controller/memcached/memcached_controller.go [hunk 3]
```diff
@@ -2,10 +2,9 @@ package memcached
 
 import (
 	"context"
+	"github.com/operator-framework/operator-sdk-samples/go/memcached-operator/pkg/apis/cache/v1alpha1"
```
Comment (estroz) [inline go/memcached-operator/pkg/controller/memcached/memcached_controller.go]: These imports should not be changed except for adding `go/` to the path.

File: go/memcached-operator/pkg/controller/add_memcached.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package controller
+
+import (
+	memcached2 "github.com/operator-framework/operator-sdk-samples/go/memcached-operator/pkg/controller/memcached"
```
Comment (estroz) [inline go/memcached-operator/pkg/controller/add_memcached.go]: Why is the import package name being changed here?

File: go/memcached-operator/pkg/apis/addtoscheme_cache_v1alpha1.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package apis
+
+import (
+	v1alpha12 "github.com/operator-framework/operator-sdk-samples/go/memcached-operator/pkg/apis/cache/v1alpha1"
```
Comment (estroz) [inline go/memcached-operator/pkg/apis/addtoscheme_cache_v1alpha1.go]: Why is the import package name being changed here?

File: go/memcached-operator/cmd/manager/main.go [hunk 6]
```diff
@@ -4,17 +4,16 @@ import (
 	"context"
 	"flag"
 	"fmt"
+	"github.com/operator-framework/operator-sdk-samples/go/memcached-operator/pkg/apis"
```
Comment (estroz) [inline go/memcached-operator/cmd/manager/main.go]: These imports should not be moved.

File: .gitignore [hunk 7]
```diff
@@ -1,3 +1,5 @@
 .idea/*
 helm/memcached-operator/.idea/*
 ansible/memcached-operator/.idea/*
+go/memcached-operator/.idea/*
```
Comment (estroz) [inline .gitignore]: You should be able to replace all of these `.idea` lines with `/*.idea/*`.

File: go/memcached-operator/.gitignore [hunk 8]
```diff
@@ -1,5 +1,5 @@
 #IDEA
-.idea/*
+.idea/workspace.xml
```
Comment (estroz) [inline go/memcached-operator/.gitignore]: The entire dir should be ignored, like in the top level `.gitignore`.

File: .gitignore (hunk 7 above)
Comment (camilamacedo86) [reply .gitignore]: it does not work.

Comment (camilamacedo86) [reply .gitignore]: done

File: go/memcached-operator/.gitignore (hunk 8 above)
Comment (camilamacedo86) [reply go/memcached-operator/.gitignore]: done

**Comment by camilamacedo86** [comment]
Hi @estroz all done. Please let me know if it is ok now?

File: go/memcached-operator/test/e2e/memcached_test.go [hunk 9]
```diff
@@ -20,9 +20,9 @@ import (
 	"testing"
 	"time"
 
-	apis "github.com/operator-framework/operator-sdk-samples/memcached-operator/pkg/apis"
-	operator "github.com/operator-framework/operator-sdk-samples/memcached-operator/pkg/apis/cache/v1alpha1"
-
+	"github.com/operator-framework/operator-sdk-samples/go/memcached-operator/pkg/apis"
+	operator "github.com/operator-framework/operator-sdk-samples/go/memcached-operator/pkg/apis/cache/operator"
```
Comment (estroz) [inline go/memcached-operator/test/e2e/memcached_test.go]: It doesn't look like `pkg/apis/cache/operator` exists.

**Pull request merged**